hoomd/simulation.py

```diff
@@ -236,12 +236,12 @@ def operations(self, operations):
                     "Cannot add `hoomd.Operations` object that belongs to "
                     "another `hoomd.Simulation` object.")
             # Switch out `hoomd.Operations` objects.
-            elif self._operations._scheduled:
+            self._operations._simulation = None
+            operations._simulation = self
+            self._operations = operations
+            if self._operations._scheduled:
                 self._operations._unschedule()
```

Comment:
Do we guarantee that `unschedule` will never reference `self._simulation`?

I would think it safer to:
```
reschedule = False
if self._operations._scheduled:
    self._operations._unschedule()
    reschedule = True

self._operations._simulation = None
operations._simulation = self
self._operations = operations

if reschedule:
    self._operations._schedule()
```

The `if reschedule` check may not be necessary as we don't necessarily need to maintain the scheduled status of the operations after setting.

Reply:
In current implementation it does not ever, but we could do that. I am just thinking it may be confusing to users to need a `run(0)` call again to do some things, but I am open to either approach.